library/misc/datafixerupper/src/main/java/org/quiltmc/qsl/datafixerupper/api/QuiltDataFixes.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.datafixerupper.api;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.DataFixerBuilder;
+import com.mojang.datafixers.schemas.Schema;
+
+import net.minecraft.nbt.NbtCompound;
+
+import org.quiltmc.qsl.datafixerupper.impl.QuiltDataFixesInternals;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Provides methods to register custom {@link DataFixer}s.
+ */
+public final class QuiltDataFixes {
+	private QuiltDataFixes() { }
+
+	/**
+	 * A "base" version 0 schema, for use by all mods.<br>
+	 * This schema <em>must</em> be the first one added!
+	 *
+	 * @see DataFixerBuilder#addSchema(int, BiFunction)
+	 */
+	public static final BiFunction<Integer, Schema, Schema> BASE_SCHEMA = (version, parent) -> {
+		checkArgument(version == 0, "version must be 0");
+		checkArgument(parent == null, "parent must be null");
+		return QuiltDataFixesInternals.createBaseSchema();
+	};
+
+	/**
+	 * Registers a new data fixer.
+	 *
+	 * @param modId the ID of the mod
+	 * @param currentVersion the current version of the mod's data
+	 * @param dataFixer the data fixer
+	 */
+	public static void registerFixer(String modId, int currentVersion, DataFixer dataFixer) {
```

Comment:
Wondering if it would be worth it to include overloads using the ModContainer class, so it creates an incentive to re-use the mod container given in the ModInitializer.